deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/impl/KubernetesDeployer.java

```diff
@@ -427,4 +428,24 @@ private void deltaWait(long milliSeconds) {
         } catch (InterruptedException e) {
         }
     }
+
+    @Override
+    public List<ReplicaInfo> getReplicas(DeploymentContext context, boolean isFilter)
+            throws HyscaleException {
+        V1PodHandler v1PodHandler = (V1PodHandler) ResourceHandlers.getHandlerOf(ResourceKind.POD.getKind());
+        String selector = ResourceSelectorUtil.getServiceSelector(context.getAppName(), context.getServiceName());
+        ApiClient apiClient = clientProvider.get((K8sAuthorisation) context.getAuthConfig());
+        List<V1Pod> podList = v1PodHandler.getBySelector(apiClient, selector, true, context.getNamespace());
+        
+        if (podList == null || podList.isEmpty()) {
+            return null;
+        }
+        
+        if (!isFilter) {
+            return K8sPodUtil.getReplicaInfo(podList);
+        }
+        
```

Comment:
K8sPodUtil is getting fat, move this to a separate class, it is more close to replicas.